kubernetes-deployment/bootstrap.py

```diff
@@ -407,9 +433,27 @@ def option_validation(args):
         if args.deploy or args.clean:
             print "You could not specify the option (-a or -r) with the option (-d or -c)"
             return False
+        if args.maintain != None:
+            print "You could not specify the option (-a or -r) with the option -m"
         # Add or remove node-list
         return True
 
+    if args.maintain != None:
+
+        if args.file == None:
+            print "Please specify the nodelist.yaml's path"
+            return False
+        if args.deploy or args.clean:
+            print "You could not specify the option (-a or -r) with the option (-d or -c)"
+            return False
+        if args.maintain == "repair":
```

Comment:
This is common to the all commands, suggest you put it out in a common place to avoid code duplication.

Reply:
The logic of ```if args.maintain == "repair":```?

Reply:
I mean we use one single parameter as "-a" for args.action, it can be "add/remove/repair/deploy/clean" instead of -a/-r/-m/-d/-c. Think about what if user specifies -a and -d, user might expect add a machine and deploy, however what we will do is only adding machines, to me looks like a little confusing.

---
In reply to: [164983160](https://github.com/Microsoft/pai/pull/281#discussion_r164983160) [](ancestors = 164983160)

Reply:
I see, in next pr do it.